src/datasets/load.py

```diff
@@ -621,6 +621,50 @@ def load_metric(
     return metric
 
 
+def load_dataset_builder(
+    path: str,
+    name: Optional[str] = None,
+    data_dir: Optional[str] = None,
+    data_files: Union[Dict, List] = None,
+    cache_dir: Optional[str] = None,
+    features: Optional[Features] = None,
+    download_config: Optional[DownloadConfig] = None,
+    download_mode: Optional[GenerateMode] = None,
+    script_version: Optional[Union[str, Version]] = None,
+    use_auth_token: Optional[Union[bool, str]] = None,
+    _return_resolved_file_path=False,
+    **config_kwargs,
+) -> DatasetBuilder:
+    # Download/copy dataset processing script
+    module_path, hash, resolved_file_path = prepare_module(
+        path,
+        script_version=script_version,
+        download_config=download_config,
+        download_mode=download_mode,
+        dataset=True,
+        return_resolved_file_path=True,
+        use_auth_token=use_auth_token,
+    )
+
+    # Get dataset builder class from the processing script
+    builder_cls = import_main_class(module_path, dataset=True)
+
+    # Instantiate the dataset builder
+    builder_instance: DatasetBuilder = builder_cls(
+        cache_dir=cache_dir,
+        name=name,
+        data_dir=data_dir,
+        data_files=data_files,
+        hash=hash,
+        features=features,
+        **config_kwargs,
+    )
+
+    if _return_resolved_file_path:
+        return builder_instance, resolved_file_path
+    return builder_instance
```

Comment:
This part is not very nice. Maybe it's better to define `base_path` as an optional attribute of `DatasetBuilder` and then in `DatasetBuilder.download_and_prepare` we can do the following:
```python
base_path = base_path if base_path is not None else self._base_path
```

Reply:
Good idea !